openapi.yaml

```diff
@@ -3049,13 +3049,25 @@ paths:
   '/jobs/{job_id}/results':
     parameters:
       - $ref: '#/components/parameters/job_id'
+      - name: partial
+        description: >-
+          If set to `true`, the results endpoint returns incomplete results while
+          still running.
+
+          Enabling this parameter requires to indicate the incomplete state of the
+          STAC metadata by setting the `openeo:incomplete` property to `true`.
+        in: query
+        allowEmptyValue: true
+        schema:
+          type: boolean
+          default: false
     get:
-      summary: Download results for a completed batch job
+      summary: Download results for a batch job
       operationId: list-results
       description: |-
-        Lists signed URLs pointing to the processed files after the batch job
+        Lists signed URLs pointing to the processed files, usually after the batch job
         has finished. Back-ends may also point to intermediate results after the
-        job has stopped due to an error.
+        job has stopped due to an error or if the `partial` parameter has been set.
```

Comment:
I wasn't aware of this "intermediate results after the job has stopped due to an error" feature

I guess it also makes sense to enforce usage the "incomplete_result" field for that case too?


Related: to me, "finished" somewhat implies "finished successfully", but the description seems to also include "failed jobs", is that correct? The new `partial` option could make the interface less error prone:
- partial=false:  only complete results from successfully finished jobs are returned, all other cases give JobNotFinished error
- partial=true: incomplete results results from failed jobs are returned

Reply:
> I guess it also makes sense to enforce usage the "incomplete_result" field for that case too?

Not sure, but at least the result should indicate the state. Maybe we should just reflect the general job status in the result (i.e. the "status" property from the job metadata), e.g. as `openeo:status` instead of using the `openeo:incomplete` flag.

> "finished" somewhat implies "finished successfully"

I guess "finished" was meant to correspond to the openEO status "finished", which would be successfully finished indeed. I'm not sure whether this is used consistently in the API though. Will need to check.


Is it okay to change the API behavior here? Strictly this is breaking, but who would be relying on "errored" results being returned anyway? :-D So I think it should be fine to change anyway.

Reply:
> Not sure, but at least the result should indicate the state.

Indeed I was also thinking along those lines: use a status enum field (e.g. finished/error/running, ..) instead of `openeo:incomplete` boolean.

Reply:
Replaced openeo:incomplete with openeo:status.